orderer/common/server/main.go

```diff
@@ -112,58 +112,91 @@ func Main() {
-				logger.Panicf("Failed validating bootstrap block: %v", err)
-			}
-			sysChanLastConfigBlock := extractSysChanLastConfig(lf, bootstrapBlock)
-			clusterBootBlock = selectClusterBootBlock(bootstrapBlock, sysChanLastConfigBlock)
-
-			typ := consensusType(bootstrapBlock, cryptoProvider)
-			clusterType = isClusterType(clusterBootBlock, cryptoProvider)
-			if clusterType {
-				logger.Infof("Setting up cluster for orderer type %s", typ)
-				clusterClientConfig = initializeClusterClientConfig(conf)
-				clusterDialer = &cluster.PredicateDialer{
-					Config: clusterClientConfig,
-				}
 
-				r = onboarding.NewReplicationInitiator(lf, bootstrapBlock, conf, clusterClientConfig.SecOpts, signer, cryptoProvider)
-				// Only clusters that are equipped with a recent config block can replicate.
-				if conf.General.BootstrapMethod == "file" {
-					r.ReplicateIfNeeded(bootstrapBlock)
-				}
+	// This happens only when the orderer is operating with a system channel.
+	if clusterBootBlock != nil {
+		consensusTypeName := consensusType(clusterBootBlock, cryptoProvider)
+		_, clusterType = clusterTypes[consensusTypeName]
 
-				if reuseGrpcListener = reuseListener(conf, typ); !reuseGrpcListener {
-					clusterServerConfig, clusterGRPCServer = configureClusterListener(conf, serverConfig, ioutil.ReadFile)
-				}
+		if clusterType {
+			logger.Infof("Setting up cluster for orderer type %s", consensusTypeName)
+			clusterClientConfig = initializeClusterClientConfig(conf)
+			clusterDialer = &cluster.PredicateDialer{
+				Config: clusterClientConfig,
+			}
 
-				// If we have a separate gRPC server for the cluster,
-				// we need to update its TLS CA certificate pool.
-				serversToUpdate = append(serversToUpdate, clusterGRPCServer)
+			if reuseGrpcListener = reuseListener(conf); !reuseGrpcListener {
+				clusterServerConfig, clusterGRPCServer = configureClusterListener(conf, serverConfig, ioutil.ReadFile)
 			}
-			// Are we bootstrapping?
-			if len(lf.ChannelIDs()) == 0 {
-				initializeBootstrapChannel(clusterBootBlock, lf)
-			} else {
-				logger.Info("Not bootstrapping because of existing channels")
+
+			// If we have a separate gRPC server for the cluster,
+			// we need to update its TLS CA certificate pool.
+			serversToUpdate = append(serversToUpdate, clusterGRPCServer)
+
+			// Are we bootstrapping with a block with number >0 ?
+			// Only clusters that are equipped with a recent config block (number i.e. >0) can replicate.
+			// This will replicate all channels if the clusterBootBlock number > system-channel height (i.e. there is a gap in the ledger).
+			repInitiator = onboarding.NewReplicationInitiator(lf, clusterBootBlock, conf, clusterClientConfig.SecOpts, signer, cryptoProvider)
+			if conf.General.BootstrapMethod == "file" {
+				repInitiator.ReplicateIfNeeded(clusterBootBlock)
 			}
 		}
+	} else {
+		logger.Infof("Setting up cluster when there is no system channel: assuming cluster-type consensus")
```

Comment:
"assuming cluster-type consensus" seems like an odd thing to include in an info message, especially when the first part of the message mentions we're setting up a cluster.

Reply:
Yes, it sounds weird. Any suggestions?

Reply:
Just seems redundant to include that last part is what I was getting at. I'd drop everything after system channel. And also reword "when there is no system channel" to "without a system channel" to be consistent with the other existing log messages.

Reply:
ok